src/main/java/ch/wisv/areafiftylan/security/authentication/AuthenticationController.java

```diff
@@ -97,6 +105,25 @@ public ModelAndView getLoginPage() {
         return createResponseEntity(HttpStatus.OK, "Token is valid!");
     }
 
+    @PreAuthorize("isAuthenticated()")
+    @RequestMapping(value = "/token/logout", method = RequestMethod.GET)
+    public ResponseEntity<?> removeSession(@RequestHeader("X-Auth-Token") String xAuth) {
+        if (!Strings.isNullOrEmpty(xAuth)) {
+            Optional<AuthenticationToken> tokenOptional = tokenRepository.findByToken(xAuth);
+            AuthenticationToken token = tokenOptional.orElseThrow(XAuthTokenNotFoundException::new);
+            if (!token.isValid()) {
```

Comment:
With this amount of logic, it may finally be time for a dedicated `AuthenticationService`